test/test-helpers.h

```diff
@@ -38,25 +36,7 @@ class TempFile {
 		std::string msg;
 	};
 
-	TempFile()
-	{
-		init_tempdir();
-		init_file();
-	}
-
-	~TempFile()
-	{
-		::unlink(filepath.c_str());
-		::rmdir(tempdir.c_str());
-	}
-
-	const std::string getPath()
-	{
-		return filepath;
-	}
-
-private:
-	void init_tempdir()
+	MainTempDir()
```

Comment:
The point was to have one directory where all of our tests can store something. It's easier to clean up by hand this way. If we use `mkdtemp` here, we won't need "main temp dir" at all, because each instance will create its own.

Do you see any downsides to using one pre-configured subdirectory to store all temporary stuff?

Reply:
Not specifically, no. But this way, we are assuming that this memory will occasionaly be cleaned up "automatically", otherwise over time, the `rand()` based loop might fail one day. This is not really an issue for tempfs-backed storages, etc.pp., and with `mkstemp` now this is even less likely to happen. But as this is not really an issue that concerns end users, but only developers, I don't think this is that much of an issue, so I'm fine with Your reasoning.